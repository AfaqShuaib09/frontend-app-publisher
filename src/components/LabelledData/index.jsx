import React from 'react';

import PropTypes from 'prop-types';

const defaultProps = {
  value: '',
  html: false,
};

const inputProps = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  html: PropTypes.bool,
};

function LabelledData({ label, value, html }) {
  return (
    <div className="row">
      <div className="labelled-data-label col-2 font-weight-bold">{label}:</div>
      {html &&
        <div
          className="labelled-data-value col"
          dangerouslySetInnerHTML={{ __html: value }} // eslint-disable-line react/no-danger
        />
      }
      {!html && <div className="labelled-data-value col">{value}</div>}
    </div>
  );
}

LabelledData.defaultProps = defaultProps;
LabelledData.propTypes = inputProps;

export default LabelledData;
