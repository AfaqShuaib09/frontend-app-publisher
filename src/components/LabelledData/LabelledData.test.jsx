import React from 'react';
import { shallow } from 'enzyme';

import LabelledData from './index';

describe('LabelledData', () => {
  it('accepts elements', () => {
    const component = shallow(<LabelledData label={<i>Hello</i>} value={<b>Goodbye</b>} />);
    expect(component).toMatchSnapshot();
  });

  it('escapes html by default', () => {
    const component = shallow(<LabelledData label="<i>1</i>" value="<b>2</b>" />);
    expect(component).toMatchSnapshot();
  });

  it('renders html if asked', () => {
    const component = shallow(<LabelledData label="<i>1</i>" value="<b>2</b>" html />);
    expect(component).toMatchSnapshot();
  });
});
