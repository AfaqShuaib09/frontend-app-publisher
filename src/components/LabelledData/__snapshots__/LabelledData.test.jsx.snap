// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledData accepts elements 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelledData
    html={false}
    label={
      <i>
        Hello
      </i>
    }
    value={
      <b>
        Goodbye
      </b>
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="labelled-data-label col-2 font-weight-bold"
        >
          <i>
            Hello
          </i>
          :
        </div>,
        false,
        <div
          className="labelled-data-value col"
        >
          <b>
            Goodbye
          </b>
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i>
              Hello
            </i>,
            ":",
          ],
          "className": "labelled-data-label col-2 font-weight-bold",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello",
            },
            "ref": null,
            "rendered": "Hello",
            "type": "i",
          },
          ":",
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <b>
            Goodbye
          </b>,
          "className": "labelled-data-value col",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Goodbye",
          },
          "ref": null,
          "rendered": "Goodbye",
          "type": "b",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="labelled-data-label col-2 font-weight-bold"
          >
            <i>
              Hello
            </i>
            :
          </div>,
          false,
          <div
            className="labelled-data-value col"
          >
            <b>
              Goodbye
            </b>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i>
                Hello
              </i>,
              ":",
            ],
            "className": "labelled-data-label col-2 font-weight-bold",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello",
              },
              "ref": null,
              "rendered": "Hello",
              "type": "i",
            },
            ":",
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              Goodbye
            </b>,
            "className": "labelled-data-value col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Goodbye",
            },
            "ref": null,
            "rendered": "Goodbye",
            "type": "b",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabelledData escapes html by default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelledData
    html={false}
    label="<i>1</i>"
    value="<b>2</b>"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="labelled-data-label col-2 font-weight-bold"
        >
          &lt;i&gt;1&lt;/i&gt;
          :
        </div>,
        false,
        <div
          className="labelled-data-value col"
        >
          &lt;b&gt;2&lt;/b&gt;
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "<i>1</i>",
            ":",
          ],
          "className": "labelled-data-label col-2 font-weight-bold",
        },
        "ref": null,
        "rendered": Array [
          "<i>1</i>",
          ":",
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<b>2</b>",
          "className": "labelled-data-value col",
        },
        "ref": null,
        "rendered": "<b>2</b>",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="labelled-data-label col-2 font-weight-bold"
          >
            &lt;i&gt;1&lt;/i&gt;
            :
          </div>,
          false,
          <div
            className="labelled-data-value col"
          >
            &lt;b&gt;2&lt;/b&gt;
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "<i>1</i>",
              ":",
            ],
            "className": "labelled-data-label col-2 font-weight-bold",
          },
          "ref": null,
          "rendered": Array [
            "<i>1</i>",
            ":",
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<b>2</b>",
            "className": "labelled-data-value col",
          },
          "ref": null,
          "rendered": "<b>2</b>",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabelledData renders html if asked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelledData
    html={true}
    label="<i>1</i>"
    value="<b>2</b>"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="labelled-data-label col-2 font-weight-bold"
        >
          &lt;i&gt;1&lt;/i&gt;
          :
        </div>,
        <div
          className="labelled-data-value col"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<b>2</b>",
            }
          }
        />,
        false,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "<i>1</i>",
            ":",
          ],
          "className": "labelled-data-label col-2 font-weight-bold",
        },
        "ref": null,
        "rendered": Array [
          "<i>1</i>",
          ":",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "labelled-data-value col",
          "dangerouslySetInnerHTML": Object {
            "__html": "<b>2</b>",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="labelled-data-label col-2 font-weight-bold"
          >
            &lt;i&gt;1&lt;/i&gt;
            :
          </div>,
          <div
            className="labelled-data-value col"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<b>2</b>",
              }
            }
          />,
          false,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "<i>1</i>",
              ":",
            ],
            "className": "labelled-data-label col-2 font-weight-bold",
          },
          "ref": null,
          "rendered": Array [
            "<i>1</i>",
            ":",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "labelled-data-value col",
            "dangerouslySetInnerHTML": Object {
              "__html": "<b>2</b>",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
